service/project.go

```diff
@@ -1009,3 +1015,87 @@ func verifyAliasExists(alias, projectId string, newAliasDefinitions []model.Proj
 	}
 	return false, nil
 }
+
+func bbProjectIsValid(projName string, proj evergreen.BuildBaronSettings) error {
+	webHook, _, _ := model.IsWebhookConfigured(projName, "")
+	webhookConfigured := webHook.Endpoint != ""
+
+	if !webhookConfigured && proj.TicketCreateProject == "" {
+		msg := "ticket_create_project and taskAnnotationSettings.FileTicketWebhook endpoint cannot both be blank"
+		grip.Critical(message.Fields{
+			"message":      msg,
+			"project_name": projName,
+		})
+		return errors.Errorf(msg)
+	}
+	if !webhookConfigured && len(proj.TicketSearchProjects) == 0 {
+		msg := "ticket_search_projects cannot be empty"
```

Comment:
We should make these a bit more user-friendly, using whatever terms the user would understand (must provide projects to search)